### DISCUSSION TITLE
Unable to install

### DISCUSSION THREAD

**Discussion by KingAab001** [ExistentialAudio/BlackHole#638] [General]
Can you take a look at it and let me know any fixes
<img width="1440" alt="Trouble shoot- Black Hole" src="https://user-images.githubusercontent.com/110447126/182302579-1f673b6b-c3e4-47f5-98d5-b1f92ce26369.png">
.

**Comment by devinroth** [comment]
Did the driver install to Library/Audio/Plug-Ins/HAL?

**Reply by KingAab001** [reply]
it is unable to install, and doesn't show up in  Library/Audio/Plug-Ins

**Comment by devinroth** [comment]
Try the home-brew method.

**Reply by KingAab001** [reply]
do you think it has anything to do with this
<img width="1440" alt="Screen Shot 2022-08-02 at 12 31 02 AM" src="https://user-images.githubusercontent.com/110447126/182317961-473805e4-28c3-4935-9acc-47619a1520da.png">
?

**Comment by devinroth** [comment]
No that’s correct.

**Reply by KingAab001** [reply]
I tries Homebrew as well, take a look, didn't wor
<img width="1440" alt="Screen Shot 2022-08-02 at 1 14 17 AM" src="https://user-images.githubusercontent.com/110447126/182326588-8ebd3d4b-e437-4a22-bd86-5d606aab803f.png">
k

**Comment by devinroth** [comment]
What version of MacOS are you on?

Does your user have limited access? You need root access to install BlackHole.

Devin

**Reply by KingAab001** [reply]
it shows the same error as root, any other problem you think it might be having?

**Comment by KingAab001** [comment]
I'm on Big Sur 11.6.8 this is what terminal says:
installer: Package name is BlackHole: Virtual Audio Driver 2ch v0.4.0
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “BlackHole2ch.v0.4.0.pkg”.)
==> Purging files for version 0.4.0 of Cask blackhole-2ch
Error: Failure while executing; `/usr/bin/sudo -E -- /usr/bin/env LOGNAME=sarkaar01 USER=sarkaar01 USERNAME=sarkaar01 /usr/sbin/installer -pkg /usr/local/Caskroom/blackhole-2ch/0.4.0/BlackHole2ch.v0.4.0.pkg -target /` exited with 1. Here's the output:
installer: Package name is BlackHole: Virtual Audio Driver 2ch v0.4.0
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “BlackHole2ch.v0.4.0.pkg”.)

**Comment by McPassion** [comment]
I am having the exact same issue:(
<img width="637" alt="Untitled" src="https://user-images.githubusercontent.com/128197816/226075741-9ad2dcbe-deb2-4cff-b193-1d1338ba40d3.png">

**Comment by devinroth** [comment]
What OS?

Try moving the installer to a different location.

Devin

**Comment by McPassion** [comment]
I only seem to have one option

<img width="935" alt="Untitled 2" src="https://user-images.githubusercontent.com/128197816/226108699-c64c7580-10b5-451f-a40d-8882295a3f3a.png">

**Comment by devinroth** [comment]
I don’t mean install to a different location. I mean move the installer pkg to your desktop and then run it.

**Reply by McPassion** [reply]
you sir, are a gentleman and a scholar. Thank you and have a great day / weekend / month / year / decade / life.

**Reply by ballanon** [reply]
I am having the same issue even after trying to install from download folder and desktop it is still not working. I will be installing an alternative pkg in the meantime but if there is a solution that I can find I will use blackhole. That being said anything new?? Thanks in advanced if you are able to help...

**Reply by nechaevdeveloper** [reply]
Moving the installer to another folder helped!!!!

**Reply by KhantheDalek** [reply]
Still having issues with this M1 Mac mini MacOS 14.4 beta. Tried all solutions named in this thread. Homebrew install, moving the .pkg to a different folder. Any other tips?

**Reply by aditalwar** [reply]
I just had a similar problem.. in my case however, files did get installed to /Library/Audio/Plug-Ins/HAL
But I was still getting the same error.. 
The way I got around to is by running this command and afterwards it adds blackhole to available devices in the audio settings - 
		`sudo sudo killall coreaudiod`

**Reply by lulu-woah** [reply]
This one works for me! I have MacOS 13.3.1. Having same issue by installing the package in download file. By moving package to desktop works well.

**Comment by Pawel-Dymek** [comment]
I'm experiencing the same problem on macOS Sonoma 14.4 on MacBook Pro with M3 Pro chip.

I've installing via HomeBrew, via regular installer, and also via said installer but moved to a few different locations, like Desktop, etc.

**Reply by luisedolozano** [reply]
I changed the location of the .pgk file from Downloads folder to Desktop and it worked. Installation successful

**Comment by devinroth** [comment]
Restart and check audio midi setup.

Devin

**Comment by gregorycron** [comment]
Won't install for me either, tried on Desktop. Running Sonoma 14.4.  I even donated the $10, this is disappointing.

**Comment by devinroth** [comment]
Restart computer and check audio midi setup. Then report back.

I think I’m going to have to change the installer to force restart computer since so many people are having issues with the post install script failing.

Devin

**Reply by gregorycron** [reply]
After restarting the Mac, it seems to work now.  Ya, I guess an auto restart or a prompt to restart would be a good thing.  Thanks for the very quick response, I wasn't expecting that.  Have a great weekend.

**Comment by devinroth** [comment]
I think Apple changed something that doesn’t allow me to reboot core audio on the latest update. A restart is annoying but I think it’s the only way.

Devin

**Reply by gchilds** [reply]
If you add an option for 14.4 to this script @devinroth, install will work without a reboot:

https://github.com/ExistentialAudio/BlackHole/blob/ed766c7d39c544196a1ab2e6ba86c79ff38fba90/Installer/scripts/postinstall#L5

the `launchctl kickstart` method now gets an `operation not permitted` error, while `killall -9` still works

**Reply by devinroth** [reply]
@gchilds does `killall -9 coreaudiod` work for every version? I know `killall coreaudiod` stopped working for some versions. And now Apple actually recommends restarting the computer. I'll give `killall -9` a try and see what feedback I get. If I still get the installer failing for some people it might be best to just force a restart.

**Comment by pemonter** [comment]
I'm not able to install it. I've tried via brew, I've tried changing the pkg location, nothing seems to work. Is there another way?

MacOS Sonoma 14.4
MacBook Pro M1 Pro

**Reply by WakeUpN3O** [reply]
I have the same issue

MacOS Sonoma 14.4
MacBook Air M1

<img width="635" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/46edce43-a23a-4791-a680-6e0ce38ed687">

<img width="709" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/b94a5585-ad49-4a3d-9609-232cef1a0232">

<img width="703" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/4a47aeab-7f57-496a-a878-87419c65f6cb">

**Comment by devinroth** [comment]
Restart your computer.

Devin

**Reply by pemonter** [reply]
Unsuccessfully. The installation isn’t completed.

**Comment by devinroth** [comment]
Restart your computer.

Devin

**Reply by WakeUpN3O** [reply]
Nothing had changed =(

**Reply by WakeUpN3O** [reply]
So guys, I relaunched "coreaudiod" process in "Activity monitor" and installed the app again. It's working now, try it.

<img width="1300" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/2c095310-a480-444a-8751-87688059e0d6">

<img width="907" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/eba0c208-c680-4fc9-a7af-2e4cf57d1274">

<img width="683" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/132017923/4c63c867-e86f-4439-af23-af135ffc903e">

**Reply by pemonter** [reply]
@WakeUpN3O how did you manage to install it? I can't finish the installation.

**Reply by devinroth** [reply]
@pemonter you misunderstand. The installation works but the post install script to restart core audio fails. Just restart CoreAudio like @WakeUpN3O suggests.

**Reply by pemonter** [reply]
@devinroth for me, the installation doesn't even work.

**Comment by devinroth** [comment]
Is BlackHole in the HAL folder?

Devin

**Reply by pemonter** [reply]
I followed @WakeUpN3O steps again and it worked now. Thanks! 👏

**Comment by devinroth** [comment]
I’ll get a new installer built this week.

Devin

**Reply by yokarra** [reply]
That would be lovely.

As i'm sure you could guess, this also happened to me and so, I to am yet another blank face, staring up at the triangular, yellow face of the immortal error message. It haunts us, but It has now become the enemy, the one and only hurdle that dares to stand between millions of people and a life of piracy, er, um, I mean a life of ethical responsible audio recording. You can not fight it alone, but I have no clue what I'm doing, like at all, so I can't um really help, much, I think. idk 👉👈

**Comment by Kucchan67346** [comment]
I tried to install the installer zip using a homemade build, 
However, due to a bug in macOS Sonoma 14.4, CoreAudio cannot be initialized, so only 2ch works.
https://forums.developer.apple.com/forums/thread/748228

This method currently only recognizes and works with BlackHole 2ch.
Since the installer is not used, the installation itself can be performed.

Please download the file below and put it in "Library/Audio/Plug-Ins/HAL/"
https://drive.google.com/file/d/139_RiQw1lEDnoXJS1uWZiqljwoagGqCG/view?usp=sharing
(If the HAL folder does not exist, please create a new one)

After that, if you restart, you should be able to use only BlackHole (2ch).

**Comment by gchilds** [comment]
I’ve used killall back to high sierra. I switched to launchctl in attempt
to be more civilised (yet still unsupported) and look where it got me.

**Comment by devinroth** [comment]
Lol.

Devin

**Reply by bytehala** [reply]
Thanks for your work, Devin. Donated!

**Comment by gchilds** [comment]
Would you look at that - this change was [actually documented and a solution proposed](https://developer.apple.com/documentation/macos-release-notes/macos-14_4-release-notes) - I am shocked. You could have knocked me down with a feather.

<img width="955" alt="image" src="https://github.com/ExistentialAudio/BlackHole/assets/2145453/181ea857-4a41-401d-8195-eb401fbb6360">

**Comment by edlime** [comment]
Experiencing the same issue with `brew` as OP.

```Error: Failure while executing; `/usr/bin/sudo -u root -E LOGNAME=xxx USER=xxx USERNAME=xxx -- /usr/sbin/installer -pkg /opt/homebrew/Caskroom/blackhole-16ch/0.5.0/BlackHole16ch.v0.5.0.pkg -target /` exited with 1. Here's the output:
installer: Package name is BlackHole: Virtual Audio Driver 16ch v0.5.0
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “BlackHole16ch.v0.5.0.pkg”.)```

**Comment by devinroth** [comment]
Brew has the wrong package. We need to update brew to 0.6.0

Devin

**Comment by gromsterus** [comment]
Workaround if you see an error when upgrading from 0.5.0 -> 0.6.0 with `brew`
```
brew upgrade
...
Error: blackhole-16ch: Failure while executing; 
`/usr/bin/sudo -E -- /bin/launchctl kickstart -kp system/com.apple.audio.coreaudiod` exited with 1. 
Here's the output:
Could not kickstart service "com.apple.audio.coreaudiod": 1: Operation not permitted
```

1. Find rb script path with `brew install --verbose --debug --cask blackhole-16ch`
Path on my mac: `/opt/homebrew/Caskroom/blackhole-16ch/.metadata/0.5.0/20230725142129.283/Casks/blackhole-16ch.rb`
2. Edit file and replace `uninstall_postflight` block with
```ruby
#  uninstall_postflight do
#    system_command "/bin/launchctl",
#                   args:         [
#                     "kickstart",
#                     "-kp",
#                     "system/com.apple.audio.coreaudiod",
#                   ],
#                   sudo:         true,
#                   must_succeed: true
#  end
  uninstall_postflight do
    system_command "/usr/bin/killall",
                   args:         ["coreaudiod"],
                   sudo:         true,
                   must_succeed: true
  end
```
3. Execute `brew upgrade` again

**Reply by sdevgill** [reply]
Yep this solved it, thanks!

**Reply by remasterzerouvm** [reply]
Worked for me too!!!
thanks

**Reply by afeicool** [reply]
Thanks!

**Reply by jetdev1** [reply]
Thanks for the solution!

**Reply by doogle-oss** [reply]
Perfect Thanks for the solution was able to move from 0.5.0 to 0.6.0

**Reply by austinarchibald** [reply]
Thank you, this worked for me too.

**Reply by alazycoder101** [reply]
Thanks, it's working for me.

**Comment by electroliquid** [comment]
For me I was finally Able to get it to work in Ventura 13.7.2 by putting it on the desktop HOLDING CTL button while opening and then at some point during the process I got a MacOS command that asked for permissions on the desktop access, and this did the trick for me finally.  I had to try it multiple times, but having it on the desktop somehow helped (I had put the .pkg file on my dropbox directory before that).  And the key is open it in the finder while HOLDING CTL.  According to Apple.  Also the link below will show what to do on other Mac OS versions if you click the selector on  Apple's website. 


[https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/13.0/mac/13.0](https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/13.0/mac/13.0)
